lib/puppet/type/mount.rb

```diff
@@ -293,5 +297,18 @@ def value(name)
       dependencies[0..-2]
     end
 
+    # Autorequire the mount point's file resource
+    autorequire(:file) { Pathname.new(@parameters[:name].value) }
+
+    # Autobefore the mount point's child file paths
+    autobefore(:file) do
+      dependencies = []
+      file_resources = catalog.resources.reject { |resource| resource.type != Puppet::Type.type(:file) }
```

Comment:
`resource.type` will return the symbolic name for the type (specified when the type is defined in Puppet::Type.newtype(:something)). For example, `:file`, `:stage`, etc.

`Puppet::Type.type(:file)` will return the `Puppet::Type::File` class that puppet's DSL generates.

So I think you'll want to say `{ |resource| resource.type != :file }` or alternatively, `collect` resources types that equal `:file`.